lib/Document/Adapter/Ghostscript.php

```diff
@@ -141,14 +146,46 @@ public function getPdf($path = null)
      */
     public function getPageCount()
     {
-        $pages = Console::exec(self::getGhostscriptCli() . " -dNODISPLAY -q -c '(" . $this->path . ") (r) file runpdfbegin pdfpagecount = quit'", null, 120);
+        $pages = Console::exec($this->buildPageCountCommand(), null, 120);
         $pages = trim($pages);
 
-        if (!is_numeric($pages)) {
+        if (! is_numeric($pages)) {
             throw new \Exception('Unable to get page-count of ' . $this->path);
         }
 
-        return (int)$pages;
+        return (int) $pages;
+    }
+
+    /**
+     * @return string
+     * @throws \Exception
+     */
+    protected function buildPageCountCommand()
+    {
+        $command = self::getGhostscriptCli() . " -dNODISPLAY -q";
+
+        if ($this->getVersion() >= 9.50) {
```

Comment:
please add a a short about version check and adding permit-file flag for better readability.